lib/operations/perform_smartconfig_request.dart

```diff
@@ -13,8 +15,21 @@ class PerformSmartconfigRequest extends AsyncPlatformOperation {
 
   @override
   Future<String> fallback() async {
-    var result = await Smartconfig.start(this.args['ssid'], this.args['bssid'], this.args['password']);
-    printDebug(result.toString());
-    return result;
+    try {
+      Map<String, dynamic> result = await Smartconfig.start(this.args['ssid'], this.args['bssid'], this.args['password']);
+      if( result != null ) {
+        // returns {246f28cce88c: 192.168.1.246}
+        debugPrint(result.toString());
```

Comment:
I think we'll be slightly better off here if we always return a Future<String> value rather than null here, so something like
```
try {
  if (result != null) {
    return Future<String>.value(result.entries.toList()[0].key);
  } else {
    return Future<String>.value("");
  }
} on PlatformException {
  return Future<String>.value("");
}
```

That'll save us needing to do any null checking in `wifi.dart`. What do you think?

Reply:
The Flutter mobile app needs to handle the common case where SmartConfig fails to get a response from a device waiting to receive SmartConfig details.   We should inform the user that SmartConfig did not connect to a device. Would they like to try again?  The program should not proceed with additional steps 3 or 4.  Let the user retransmit via SmartConfig.  Stay on Step 2.

Here's another scenario we need to handle:
Since my wifi password is non-trivial, I made a mistake entering it. (it would be nice to have an option to show the password)  What happened next was very confusing...  It passed the SSID and incorrect password to my ESP32.  The ESP32 tried to connect to the wifi network (and failed because the password was wrong) and entered an inconsistent state.  I fixed this behavior in the firmware v1.2.0.   The ESP32, if it fails to connect to the wifi, will now re-enter SmartConfig listen mode.  The Flutter app should be used to correct / send a correct wifi password.  Currently this is problematic. The password string seems to be cached by the form?  Even if I, as the user, edited the flutter form with the corrected password, it didn't work. (although I had multiple apps breaking simultaneously)   I had to EXIT the Flutter app to change the wifi password I had entered.

I don't know much about asynchronous functions.  I'm reading through the docs now.  Can't really make a recommendation about Future <String>    Learning more here: https://dart.dev/codelabs/async-await

The most important thing is that the Flutter mobile app present the user with some type of status.  "Transmitting SmartConfig details..." spinner (and a cancel button?).    "Failed to detect a device.  [Retry] [Change password] "

Is there a way to go backwards?  
Step 3 to Step 2 (so I can try a different wifi password - although the user might not know they've flubbed the password sent to the device.)  
Step 4 to Step 3 so they can fix their name / email.

Reply:
I should probably take most of the above conversation over to the OpenEEW Slack #app channel.